src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2741,6 +2753,73 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - Creates a settings UI tab and focuses it. If there's already a settings UI tab open,
+    //   just focus the existing one.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TerminalPage::_OpenSettingsUI()
+    {
+        // If we're holding the settings tab's switch command, don't create a new one, switch to the existing one.
+        if (!_switchToSettingsCommand)
+        {
+            winrt::Microsoft::Terminal::Settings::Editor::MainPage sui{ _settings };
+            if (_hostingHwnd)
+            {
+                sui.SetHostingWindow(reinterpret_cast<uint64_t>(*_hostingHwnd));
+            }
+
+            sui.OpenJson([weakThis{ get_weak() }](auto&& /*s*/, winrt::Microsoft::Terminal::Settings::Model::SettingsTarget e) {
+                if (auto page{ weakThis.get() })
+                {
+                    page->_LaunchSettings(e);
+                }
+            });
+
+            auto newTabImpl = winrt::make_self<SettingsTab>(sui);
+            _MakeSwitchToTabCommand(*newTabImpl, _tabs.Size());
+
+            // Add the new tab to the list of our tabs.
+            _tabs.Append(*newTabImpl);
+            _mruTabActions.Append(newTabImpl->SwitchToTabCommand());
+
+            newTabImpl->SetDispatch(*_actionDispatch);
+
+            // Give the tab its index in the _tabs vector so it can manage its own SwitchToTab command.
+            _UpdateTabIndices();
+
+            // Don't capture a strong ref to the tab. If the tab is removed as this
+            // is called, we don't really care anymore about handling the event.
+            auto weakTab = make_weak(newTabImpl);
+
+            auto tabViewItem = newTabImpl->TabViewItem();
+            _tabView.TabItems().Append(tabViewItem);
```

Comment:
A lot of this seems similar to the setup we do when we create a terminal tab, should we make a card to unify it?

Reply:
This was some of the code that Leon wrote up. I'm a bit worried that I may break some tab switching functionality in the process of moving stuff around. For now, I've lumped it into #8452 since that seems to be touching some similar code. I'll do it as a follow-up to this PR since this is already a pretty big one.